blmath/geometry/convexify.py

```diff
@@ -12,19 +18,49 @@ def convexify_planar_curve(polyline, flatten=False, want_vertices=False):
         - want_vertices:
             Boolean; do you want the indices to the convex hull vertices?
     '''
-    import numpy as np
-    from scipy.spatial import ConvexHull  # First thought this warning was caused by a pythonpath problem, but it seems more likely that the warning is caused by scipy import hackery. pylint: disable=no-name-in-module
-    from blmath.geometry import Polyline
-    from blmath.geometry.transform.rotation import rotate_to_xz_plane
-    from blmath.geometry.transform.translation import translation
+    if flatten:
+        return convexify_and_flatten_planar_curve(polyline, want_vertices, normal)
+
+    v = polyline.v
+    if len(v) <= 1:
+        return polyline
+
+    if normal is None:
+        normal = estimate_normal(v)
+
+    # ConvexHull(v) requires that v be projected to a plane and made N x 2.
+    # This is done by simply dropping whichever dimension is closest to the normal.
+    dim_to_drop = np.argmax(np.abs(normal))
```

Comment:
This assumes that the plane is near axis-aligned... The right way to do this would be to project the points onto a `Plane` object. Add something like this to Plane maybe:
```py
def project_points2d(self, points):
    from blmath.geometry.transform.rotation import euler
    # get a 2d basis:
    if np.all(self.normal == np.array([0., 1., 0.])):
        some_point_on_plane = self.project_point(self.reference_point + np.array([1., 0., 0.]))
    else:
        some_point_on_plane = self.project_point(self.reference_point + np.array([0., 1., 0.]))
    e0 = vx.normalize(some_point_on_plane - self.reference_point)
    e1 = np.cross(e0, self.normal)
    proj_3d = self.project_point(points)
    return np.array([vx.sproj(proj_3d, e0), vx.sproj(proj_3d, e1)])
```
then you can
```py
proj_v = Planen(np.mean(v, axis=1), normal).project_points2d(v)
```